### ISSUE TITLE
Upgrade from v0.12.25 to v0.13 problem with decoded from the state

### ISSUE THREAD

**Issue by omerh** [hashicorp/terraform#25924]
### Terraform Version

```
Terraform v0.13.0
+ provider registry.terraform.io/-/aws v3.2.0
+ provider registry.terraform.io/hashicorp/aws v3.2.0
```

Trying to run a plan from version 0.12 with version 0.13.
Backend state is s3 with encryption

```
terraform {
  backend "s3" {
    bucket  = "-------------------------"
    key     = "-----------------/----------------------.tfstate"
    region  = "us-east-1"
    encrypt = true
    profile = "mm"
  }
}

```

I ran `terraform init` and all went well

When running the plan I get an error:

```
Error: Invalid resource instance data in state

  on ../../modules/aws/eks/nodes_role/main.tf line 69:
  69: resource "aws_iam_instance_profile" "eks_nodes" {

Instance module.eks_role.aws_iam_instance_profile.eks_nodes data could not be
decoded from the state: unsupported attribute "roles".
```

**Comment by mildwonkey** [comment]
Hi @omerh , thanks for reporting this! It is a duplicate of #25752, and the fix will be in the next terraform release!

**Comment by omerh** [comment]
Thanks @mildwonkey please note that the version on #25752 is on version v0.13.0-rc1 and I am on version v0.13.0

**Comment by mildwonkey** [comment]
Yes - the problem remained in v0.13.0 and the fix is in terraform v0.13.1, which we plan on releasing next week! Thanks and sorry!
